test/common-tests/request-field-validation-api-tests/string-field-validation-api-tests.ts

```diff
@@ -0,0 +1,257 @@
+  generateFieldValueThatDoesNotMatchEnum?: () => RequestBodyItem[RequestBodyItemKey];
+  validRequestBody: RequestBodyItem[] | RequestBodyItem;
+  makeRequest: ((body: unknown[]) => request.Test) | ((body: unknown) => request.Test);
+  givenAnyRequestBodyWouldSucceed: () => void;
+}
+
+export function withStringFieldValidationApiTests<RequestBodyItem, RequestBodyItemKey extends keyof RequestBodyItem>({
+  fieldName: fieldNameSymbol,
+  length: lengthOption,
+  minLength: minLengthOption,
+  maxLength: maxLengthOption,
+  required,
+  pattern,
+  enum: theEnum,
+  generateFieldValueThatDoesNotMatchEnum,
+  generateFieldValueOfLength,
+  generateFieldValueThatDoesNotMatchRegex,
+  validRequestBody,
+  makeRequest,
+  givenAnyRequestBodyWouldSucceed,
+}: StringFieldValidationApiTestOptions<RequestBodyItem, RequestBodyItemKey>): void {
+  const fieldName = fieldNameSymbol.toString();
+  const { minLength, maxLength } = getMinAndMaxLengthFromOptions({ fieldName, minLengthOption, maxLengthOption, lengthOption, theEnum });
+  const requestIsAnArray = Array.isArray(validRequestBody);
+  const requestBodyItem = requestIsAnArray ? validRequestBody[0] : validRequestBody;
+
+  required = required ?? true;
+
+  describe(`${fieldName} validation`, () => {
+    beforeEach(() => {
+      givenAnyRequestBodyWouldSucceed();
+    });
+
+    it(`returns a 400 response if ${fieldName} is number`, async () => {
+      const requestWithNumberField = { ...requestBodyItem, [fieldNameSymbol]: 1 };
+      const preparedRequestWithNumberField = prepareModifiedRequest(requestIsAnArray, requestWithNumberField);
+
+      const { status, body } = await makeRequest(preparedRequestWithNumberField);
+
+      expect(status).toBe(400);
+      expect(body).toMatchObject({
+        error: 'Bad Request',
+        message: expect.arrayContaining([`${fieldName} must be a string`]),
+        statusCode: 400,
+      });
+    });
+
+    if (required) {
+      if (theEnum && generateFieldValueThatDoesNotMatchEnum) {
+        it(`returns a 400 response if ${fieldName} is not present`, async () => {
+          const { [fieldNameSymbol]: _removed, ...requestWithoutTheField } = requestBodyItem;
+          const preparedRequestWithoutTheField = prepareModifiedRequest(requestIsAnArray, requestWithoutTheField);
+
+          const { status, body } = await makeRequest(preparedRequestWithoutTheField);
+
+          expect(status).toBe(400);
+          expect(body).toMatchObject({
+            error: 'Bad Request',
+            message: expect.arrayContaining([`${fieldName} must be one of the following values: ${Object.values(theEnum).join(', ')}`]),
+            statusCode: 400,
+          });
+        });
+      } else {
+        it(`returns a 400 response if ${fieldName} is not present`, async () => {
+          const { [fieldNameSymbol]: _removed, ...requestWithoutTheField } = requestBodyItem;
+
+          const { status, body } = await makeRequest([requestWithoutTheField]);
```

Comment:
I noticed a small mistake in TFS this morning: this should be `makeRequest(prepareModifiedRequest(requestWithoutTheField))` instead

Can you fix it here please @avaitonis and I'll fix it in TFS?

Reply:
Fixed